core/src/window.cpp

```diff
@@ -101,4 +101,30 @@ void Window::close() {
     glfwSetWindowShouldClose(m_window, true);
 }
 
+void Window::on_awake() {
+
+}
+
+void Window::on_update(double m_delta_time) {
+    update();
+}
+
+void Window::on_late_update() {
+
+}
+
+void Window::on_destroy() {
+
+}
+
+void Window::calculate_delta_time() {
+    m_currentFrame = glfwGetTime();
```

Comment:
Do we need to store `m_currentFrame`? Because `m_lastFrame` gets this value, we could instead make scope this variable to the function.

Reply:
this is valid c:

Reply:
done